lib/timers.js

```diff
@@ -424,80 +426,148 @@ exports.enroll = function(item, msecs) {
  * DOM-style timers
  */
 
+class Timeout {
+  constructor(callback, after, args, isRepeat) {
```

Comment:
Can `args` and `isRepeat` go into an options object? It is foreseeable that: a) we may expose the constructor, and b) that we may want to also have an `unrefed` option.

Reply:
They could but then we're creating an extra object just to pass them to the constructor (which just reassigns them anyway). I would maybe leave that change to when we actually do decide to expose as it will have a small performance implication? Just thinking out loud here.